hercules-ci-agent/hercules-ci-agent/Hercules/Agent/Cachix/Info.hs

```diff
@@ -0,0 +1,32 @@
+module Hercules.Agent.Cachix.Info where
+
+import           Protolude
+import qualified Cachix.Client.Config          as Cachix.Config
+import qualified Cachix.Client.Env             as Cachix.Env
+import           Hercules.Agent.Log
+import           Hercules.Agent.Cachix.Env
+import qualified Katip.Core                    as K
+
+getAvailablePushCaches :: (MonadReader r m, HasEnv r) => m [Text]
+getAvailablePushCaches = do
+  env <- asks (cachixEnv . getEnv)
+  let
+    pushableName Cachix.Config.BinaryCacheConfig { secretKey = _required, name = name }
```

Comment:
I guess this is not done yet? :)

Reply:
`_required` is mostly there for the compiler to notify if the field is removed, or renamed, because that might indicate a problem. I guess I should add a type signature to make that work, but the real solution is to improve the cachix library so it exposes a function that does exactly what this function does :)